gsy_framework/sim_results/electric_blue/time_series.py

```diff
@@ -0,0 +1,148 @@
+
+    return result
+
+
+class ForwardDeviceTimeSeries:
+    """Generate time series data using ForwardDeviceStats object."""
+    def __init__(
+            self, device_stats: ForwardDeviceStats, product_type: AvailableMarketTypes):
+
+        self.device_stats = device_stats
+        self.product_type = product_type
+
+    def generate(self, resolution: Duration):
+        """Generate all time series."""
+        return {
+            "matched_buy_orders_kWh": self._generate_matched_buy_orders_kWh(resolution),
+            "matched_sell_orders_kWh": self._generate_matched_sell_orders_kWh(resolution),
+            "open_sell_orders_kWh": self._generate_open_sell_orders_kWh(resolution),
+            "open_buy_orders_kWh": self._generate_open_buy_orders_kWh(resolution),
+            "all_buy_orders_KWh": self._generate_all_buy_orders_kWh(resolution),
+            "all_sell_orders_kWh": self._generate_all_sell_orders(resolution)
+        }
+
+    def _generate_matched_buy_orders_kWh(self, resolution: Duration) -> Dict:
+        """Generate time series for buy trades."""
+        if self.device_stats.total_energy_bought <= 0:
+            return {}
+
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=self.device_stats.total_energy_bought)
+        matched_buy_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=self.device_stats.total_energy_bought,
+            market_slot=DateTime.fromisoformat(self.device_stats.time_slot),
+            product_type=self.product_type
+        )
+        return resample_data(matched_buy_profile, resolution, aggregator_fn=sum)
+
+    def _generate_matched_sell_orders_kWh(self, resolution: Duration) -> Dict:
+        """Generate time series for sell trades."""
+        if self.device_stats.total_energy_sold <= 0:
+            return {}
+
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=self.device_stats.total_energy_sold)
+        matched_sell_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=self.device_stats.total_energy_sold,
+            market_slot=DateTime.fromisoformat(self.device_stats.time_slot),
+            product_type=self.product_type
+        )
+        return resample_data(matched_sell_profile, resolution, aggregator_fn=sum)
+
+    def _generate_open_sell_orders_kWh(self, resolution: Duration) -> Dict:
+        """Generate time series for all offers."""
+        if not self.device_stats.open_offers:
+            return {}
+        total_offered_energy = sum([offer["energy"] for offer in self.device_stats.open_offers])
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=total_offered_energy)
+        open_sell_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=total_offered_energy,
+            market_slot=DateTime.fromisoformat(self.device_stats.time_slot),
```

Comment:
I think that it is better to unify our datetime conversions, in order to use the same methods throughout our codebase. Therefore how about replacing this with `str_to_pendulum_datetime` ?

Reply:
Regarding the other discussion in [this](https://github.com/gridsingularity/gsy-e/pull/1530#discussion_r970419992) PR, I'm about to convert them all to pendulum DateTime objects.

Reply:
Fixed.